components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -274,13 +279,46 @@ bool CosmeticFiltersJSHandler::OnIsFirstParty(const std::string& url_string) {
 void CosmeticFiltersJSHandler::OnAddSiteCosmeticFilter(
     const std::string& selector) {
   const auto host = url_.host();
-  auto handler = MakeCosmeticFiltersHandler(render_frame_);
-  handler->AddSiteCosmeticFilter(selector);
+  GetElementPickerRemoteHandler()->AddSiteCosmeticFilter(selector);
 }
 
 void CosmeticFiltersJSHandler::OnManageCustomFilters() {
-  auto handler = MakeCosmeticFiltersHandler(render_frame_);
-  handler->ManageCustomFilters();
+  GetElementPickerRemoteHandler()->ManageCustomFilters();
+}
+
+mojo::AssociatedRemote<cosmetic_filters::mojom::CosmeticFiltersHandler>&
+CosmeticFiltersJSHandler::GetElementPickerRemoteHandler() {
+  if (!element_picker_actions_handler_ ||
+      !element_picker_actions_handler_.is_bound() ||
+      !element_picker_actions_handler_.is_connected()) {
+    element_picker_actions_handler_ = MakeCosmeticFiltersHandler(render_frame_);
+    element_picker_actions_handler_.set_disconnect_handler(
+        base::BindOnce(&CosmeticFiltersJSHandler::OnRemoteHandlerDisconnect,
+                       weak_ptr_factory_.GetWeakPtr()));
+  }
+  return element_picker_actions_handler_;
+}
+
+void CosmeticFiltersJSHandler::OnRemoteHandlerDisconnect() {
+  element_picker_actions_handler_.reset();
+}
+
+void CosmeticFiltersJSHandler::GetCosmeticFilterThemeInfo() {
+  GetElementPickerRemoteHandler()->GetElementPickerThemeInfo(
+      base::BindOnce(&CosmeticFiltersJSHandler::OnGetCosmeticFilterThemeInfo,
+                     weak_ptr_factory_.GetWeakPtr()));
+}
+
+void CosmeticFiltersJSHandler::OnGetCosmeticFilterThemeInfo(
+    bool is_dark_mode_enabled,
+    int32_t background_color) {
+  blink::WebLocalFrame* web_frame = render_frame_->GetWebFrame();
+  std::string new_selectors_script = base::StringPrintf(
+      kSePickerThemeInfoScript, is_dark_mode_enabled, background_color);
+  web_frame->ExecuteScriptInIsolatedWorld(
```

Comment:
It is possible to pass the result via `getElementPickerThemeInfo(callback)` instead of using `ExecuteScriptInIsolatedWorld()`?